src/app/modules/main/chat_section/module.nim

```diff
@@ -614,14 +618,18 @@ method onCommunityCategoryChannelChanged*(self: Module, channelId: string, newCa
     else:
       self.setActiveItem(channelId)
 
-method onReorderChat*(self: Module, chatId: string, position: int, newCategoryIdForChat: string) =
+method onReorderChat*(self: Module, chatId: string, position: int, newCategoryIdForChat: string, prevCategoryId: string, prevCategoryDeleted: bool) =
   var newCategoryName = ""
   var newCategoryPos = -1
   if newCategoryIdForChat != "":
     let newCategory = self.controller.getCommunityCategoryDetails(self.controller.getMySectionId(), newCategoryIdForChat)
     newCategoryName = newCategory.name
     newCategoryPos = newCategory.position
   self.view.chatsModel().reorderChatById(chatId, position, newCategoryIdForChat, newCategoryName, newCategoryPos)
+  if prevCategoryId != "" and not prevCategoryDeleted:
+    if not self.view.chatsModel().hasEmptyChatItem(CATEGORY_ID_PREFIX & prevCategoryId):
+      let category = self.controller.getCommunityCategoryDetails(self.controller.getMySectionId(), prevCategoryId)
+      self.addEmptyChatItemForCategory(category)
```

Comment:
This is necessary because when `onReorderChat` is called, there's a case when:

a) A chat was moved out of a category, hence we need to add an empty chat items in case non exists, otherwise the category disappears
b) A category was deleted, which caused a chat to have a change in its category id, which will trigger this function as well, so we also need to check whether or not the previous category was deleted.